### PR TITLE
cli: (program) enhance test_cli_program_set_buffer_authority

### PR DESCRIPTION
**Pull request by norwnd**
#### Problem

Suboptimal coverage (in unit-tests) for changing program buffer authority.

#### Summary of Changes

This is a spin-off from https://github.com/solana-labs/solana/pull/33860 (see https://github.com/solana-labs/solana/pull/33860) to expand test coverage by checking that old authority no longer has control over buffer once buffer authority has been changed.

### CODE REVIEW

File: cli/tests/program.rs [hunk 1]
```diff
@@ -1229,12 +1229,29 @@ fn test_cli_program_set_buffer_authority() {
...
+    config.output_format = OutputFormat::JsonCompact;
+    process_command(&config).unwrap_err();
+
+    // Set buffer authority back to original authority
     config.signers = vec![&keypair, &new_buffer_authority];
     config.command = CliCommand::Program(ProgramCliCommand::SetBufferAuthority {
         buffer_pubkey: buffer_keypair.pubkey(),
         buffer_authority_index: Some(1),
-        new_buffer_authority: buffer_keypair.pubkey(),
+        new_buffer_authority: keypair.pubkey(),
```
Comment (norwnd) [inline cli/tests/program.rs:1254]: Changing `buffer_keypair ` -> `keypair` here to make it a bit more clear (separate keys serving as **authorities** from keys serving as buffers/programs/etc).

Comment (joncinque) [reply cli/tests/program.rs:1254]: It was deliberate to set the authority to the buffer itself, as evidenced by the comment `// Set authority to buffer`. There are often programmer bugs if someone tries this behavior, and we need to support it, which is why the test is there. Can you revert that part back?

**Review by joncinque — COMMENTED** [review]
Just the one comment about reverting the self-owned buffer, but after that it looks good!

File: cli/tests/program.rs (hunk 1 above)
Comment (norwnd) [reply cli/tests/program.rs]: Interesting, reverted in https://github.com/solana-labs/solana/pull/34316/commits/71972c2545df46e6ab1df387c71fad91715bc281 (also added some clarifying comments).

**Review by joncinque — APPROVED** [review]
Looks great, thanks!

**Pull request merged**